### PR TITLE
fix: demand `--force` to re-install stable browser channels

### LINKED ISSUE #13061: [BUG] `npx playwright install chrome` is dangerous
I followed an advice to run `npx playwright install chrome` on my main machine and became very nervous when I saw this line:

```
rm -rf '/Applications/Google Chrome.app'
```

If playwright tampers with my `/Applications` (which even requires `sudo`), at the very minimum ask me to confirm.

### PR DESCRIPTION
**Pull request by aslushnikov**
This patch will check if browser channel is already installed.
If it is, it'll abort installation with the following error:

```
aslushnikov:~/prog/playwright$ npx playwright install msedge
Failed to install browsers
Error:
╔═════════════════════════════════════════════════════════════════╗
║ ATTENTION: "msedge" is already installed on the system!         ║
║                                                                 ║
║ "msedge" installation is not hermetic; installing newer version ║
║ requires *removal* of a current installation first.             ║
║                                                                 ║
║ To *uninstall* current version and re-install latest "msedge":  ║
║                                                                 ║
║ - Close all running instances of "msedge", if any               ║
║ - Use "--force" to install browser:                             ║
║                                                                 ║
║     npx playwright install --force msedge                       ║
║                                                                 ║
║ <3 Playwright Team                                              ║
╚═════════════════════════════════════════════════════════════════╝
```

To re-install browser channel, use `--force`.

Fixes https://github.com/microsoft/playwright/issues/13061

### CODE REVIEW

File: packages/playwright-core/src/utils/registry.ts [hunk 1]
```diff
@@ -491,7 +491,25 @@ export class Registry {
...
       validateHostRequirements: () => Promise.resolve(),
-      _install: install,
+      _install: install ? async (forceReinstall: boolean) => {
+        const execPath = executablePath('', false);
+        if (!forceReinstall && execPath) {
+          const { langName } = getClientLanguage();
+          const command = buildPlaywrightCLICommand(langName, 'install --force ' + name);
+          throw new Error('\n' + wrapInASCIIBox([
+            `${name} is already installed on the system.`,
+            `To re-install ${name}:`,
```
Comment (mxschmitt) [inline packages/playwright-core/src/utils/registry.ts:501]: maybe good to outline there that a recent version gets installed and its not just a re-install.

File: packages/playwright-core/src/utils/registry.ts [hunk 2]
```diff
@@ -491,7 +491,25 @@ export class Registry {
       executablePathOrDie: (sdkLanguage: string) => executablePath(sdkLanguage, true)!,
       installType: install ? 'install-script' : 'none',
       validateHostRequirements: () => Promise.resolve(),
-      _install: install,
+      _install: install ? async (forceReinstall: boolean) => {
```
Comment (dgozman) [inline packages/playwright-core/src/utils/registry.ts:494]: Let's extract a function.

File: packages/playwright-core/src/utils/registry.ts [hunk 3]
```diff
@@ -491,7 +491,25 @@ export class Registry {
...
+        const execPath = executablePath('', false);
+        if (!forceReinstall && execPath) {
+          const { langName } = getClientLanguage();
+          const command = buildPlaywrightCLICommand(langName, 'install --force ' + name);
+          throw new Error('\n' + wrapInASCIIBox([
+            `${name} is already installed on the system.`,
+            `To re-install ${name}:`,
+            ``,
+            `- close all running instances of ${name}, if any`,
+            `- use "--force" to install browser`,
```
Comment (dgozman) [inline packages/playwright-core/src/utils/registry.ts:504]:
```suggestion
            `- use "--force" to install browser:`,
```

Also, be more scary in the message.

File: packages/playwright-core/src/utils/registry.ts [hunk 4]
```diff
@@ -254,6 +254,7 @@ export interface Executable {
 interface ExecutableImpl extends Executable {
   _install?: () => Promise<void>;
   _dependencyGroup?: DependencyGroup;
+  _isHermeticInstallation?: boolean;
```
Comment (dgozman) [inline packages/playwright-core/src/utils/registry.ts:257]: Probably should drop the question mark?

**Review by dgozman — APPROVED** [review]

**Pull request merged**